File: code/python/twisted/src/examples/google-test-xmlrpc.rb
require "xmlrpc/client"

# Make an object to represent the XML-RPC server.
server = XMLRPC::Client.new( "localhost", "/selenium-driver/RPC2", 8080)

# Bump timeout a little higher than the default 5 seconds
server.call('setTimeout',15)

system 'start run_firefox.bat'

puts server.call('open','http://localhost:8080/AUT/000000A/http/www.google.com/')
puts server.call('verifyTitle','Google')
puts server.call('type','q','Selenium ThoughtWorks')
puts server.call('verifyValue','q','Selenium ThoughtWorks')
puts server.call('clickAndWait','btnG')
puts server.call('verifyTextPresent','selenium.thoughtworks.com','')
puts server.call('verifyTitle','Google Search: Selenium ThoughtWorks')
puts server.call('testComplete')

File: code/python/twisted/src/examples/google-test-xmlrpc.pl
# Tested using ActiveState 5.8 perl (on Windows XP) + Frontier-RPC library
# To get and install Frontier:
# c:\> ppm
# ppm> install Frontier-RPC

use Frontier::Client;

# Make an object to represent the XML-RPC server.
$server_url = 'http://localhost:8080/selenium-driver/RPC2';
$server = Frontier::Client->new( 'url' => $server_url);

# Bump timeout a little higher than the default 5 seconds
$server->call('setTimeout',15);

system('start run_firefox.bat');

print $server->call('open','http://localhost:8080/AUT/000000A/http/www.google.com/');
print "\n";
print $server->call('verifyTitle','Google');
print "\n";
print $server->call('type','q','Selenium ThoughtWorks');
print "\n";
print $server->call('verifyValue','q','Selenium ThoughtWorks');
print "\n";
print $server->call('clickAndWait','btnG');
print "\n";
print $server->call('verifyTextPresent','selenium.thoughtworks.com','');
print "\n";
print $server->call('verifyTitle','Google Search: Selenium ThoughtWorks');
print "\n";
print $server->call('testComplete');
print "\n";